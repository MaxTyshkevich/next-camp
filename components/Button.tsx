import Image from 'next/image';

type varialts =
  | 'btn_white'
  | 'btn_white_text'
  | 'btn_green'
  | 'btn_dark_green'
  | 'btn_dark_green_outline';

type ButtonProps = {
  type: 'button' | 'submit';
  title: string;
  icon?: string;
  variant: varialts;
};

const Button = ({ type, title, icon, variant }: ButtonProps) => {
  return (
    <button
      type={type}
      className={`flexCenter rounded-full border gap-3 ${variant}`}
    >
      {icon && <Image src={icon} height={24} width={24} alt="menu" />}
      <label className="bold-16 whitespace-nowrap">{title}</label>
    </button>
  );
};

export default Button;
