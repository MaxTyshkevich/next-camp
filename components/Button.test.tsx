import { render, screen } from '@testing-library/react';
import Button, { ButtonProps } from './Button';

const validProps = {
  type: 'button',
  title: 'Testing title',
  variant: 'btn_white',
} as ButtonProps;

const fullProps = {
  ...validProps,

  full: true,
} as ButtonProps;

describe('testing Button', () => {
  /* it('testing title', () => {
    render(<Button {...props} />);

    const labelText = screen.getByLabelText(props.title);

    console.log({ labelText });
    expect(labelText).toBeInTheDocument();
  });
 */
  it('testing Button type', () => {
    render(<Button {...validProps} />);

    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button.type).toBe(validProps.type);
  });

  it('testing Button className', () => {
    const { container } = render(<Button {...validProps} />);

    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button).toHaveClass(validProps.variant);
  });

  it('testing Button className not prop full', () => {
    render(<Button {...validProps} />);

    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button).not.toHaveClass('w-full');
  });

  it('testing Button className prop full', () => {
    render(<Button {...fullProps} />);

    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button).toHaveClass('w-full');
  });

  it('Button Snapshot', () => {
    const { container } = render(<Button {...validProps} />);

    expect(container).toMatchSnapshot();
  });
});
