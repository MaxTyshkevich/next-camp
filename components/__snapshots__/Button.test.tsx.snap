// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button Button Snapshot 1`] = `
<div>
  <button
    class="flexCenter rounded-full
       border gap-3 btn_white "
    type="button"
  >
    <label
      class="bold-16 whitespace-nowrap cursor-pointer"
    >
      Testing title
    </label>
  </button>
</div>
`;
